ext/RMagick/rmimage.c

```diff
@@ -42,28 +42,34 @@ static const char *BlackPointCompensationKey = "PROFILE:black-point-compensation
 
 
 /**
- * Checks if opacity_or_alpha is a named argument called alpha and returns the alpha value or
- * converts the unnamed opacity value to alpha.
+ * Converts the opacity value to alpha and raises a warning.
  *
  * No Ruby usage (internal function)
  *
- * @opacity_or_alpha an opacity or a named alpha value
+ * @opacity an opacity value
  * @argument_name the name of the argument
  */
 static Quantum
-get_named_alpha_value(VALUE opacity_or_alpha, const char *argument_name)
+get_alpha_from_opacity(VALUE opacity, const char *argument_name)
 {
-    VALUE alpha;
-
-    if (TYPE(opacity_or_alpha) != T_HASH)
-    {
-        VALUE method = rb_id2str(rb_frame_this_func());
-        rb_warning("Image#%"PRIsVALUE" requires a named argument for '%s' and now expects an alpha value instead of an opacity value.", method, argument_name);
+    VALUE method = rb_id2str(rb_frame_this_func());
+    rb_warning("Image#%"PRIsVALUE" requires a named argument for '%s' and now expects an alpha value instead of an opacity value.", method, argument_name);
 
-        return QuantumRange - APP2QUANTUM(opacity_or_alpha);
-    }
+    return QuantumRange - APP2QUANTUM(opacity);
+}
 
-    alpha = rb_hash_aref(opacity_or_alpha, ID2SYM(rb_intern(argument_name)));
+/**
+ * Returns the alpha value from the hash.
+ *
+ * No Ruby usage (internal function)
+ *
+ * @hash the hash
+ * @argument_name the name of the argument
+ */
+static Quantum
+get_alpha_from_hash(VALUE hash, const char *argument_name)
```

Comment:
Hash keys should be fixed with `alpha` ? 
If so, `argument_name` is unnecessary.

Reply:
In a later PR the name will be different so that is why we cannot just use `alpha` and remove `argument_name `.

Reply:
Should you provide a consistent interface?

Reply:
I don't understand what you are asking me?

Reply:
I think the user is likely to make a mistake if the hash key changes at some interfaces.

Reply:
Some now have a different name in the API:
`black_threshold(red_channel, green_channel, blue_channel, alpha_channel)`

Maybe we should rename these variables so we can use `alpha` in all places?

Reply:
It is OK with another PR 😉